src/Blef.GameLogic/PokerHands/HighCard.cs

```diff
@@ -6,33 +6,26 @@ namespace Blef.GameLogic.PokerHands
 {
     public class HighCard : PokerHand
     {
+        public Card Card { get; private set; }
+
         public HighCard(Card card)
         {
             Card = card;
         }
 
-        public Card Card { get; }
-
         public override bool IsOnTable(IEnumerable<Card> cards)
         {
             return cards.Contains(Card);
         }
 
-        public override short GetRank()
+        protected override short GetRank()
         {
             return 1;
         }
 
-        public override int CompareToWithinSameRank(PokerHand value)
+        protected override int GetInnerRank()
         {
-            HighCard highCard = value as HighCard;
-
-            if (highCard == null)
-            {
-                throw new InvalidOperationException($"Can compare only {nameof(PokerHand)} with the same rank");
-            }
-
-            return Card - highCard.Card;
+            return (int)Card.Rank;
```

Comment:
Now I see that my implementation and idea were completely not clear. This whole will be changed to something else